data_ocean/views.py

```diff
@@ -44,9 +44,9 @@ def get(self, request):
 class RegisterView(viewsets.ReadOnlyModelViewSet):
     queryset = Register.objects.all()
     serializer_class = RegisterSerializer
-    filterset_class = RegisterFilter
+    filterset_class = RegisterFilterSet
     filter_backends = [DjangoFilterBackend, SearchFilter]
-    search_fields = ['name', 'source_register_id']
+    search_fields = ['name', 'source_register_id', 'source_name']
```

Comment:
I think you only need to leave only "name".
its better as search.
```suggestion
    search_fields = ['name']
```